logzio-sender/src/main/java/io/logz/sender/DiskQueue.java

```diff
@@ -65,14 +63,16 @@ public boolean isEmpty() {
         return logsBuffer.isEmpty();
     }
 
-    private void isEnoughSpace() {
+    private void validateEnoughSpace() {
```

Comment:
You need to wrap the code inside this method with
```
try {
} catch (Throwable t) {
   statusReporter.error("Failed validating we have enough space", t)
}
```

Otherwise, upon an exception, it will stop scheduling and will never run again. Read `scheduleWithFixedDelay` javadoc please.

Like you did in `gcBigQueue()`